great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -1750,161 +2279,216 @@ def _filter(
 
         return column_based_expectation_configurations_by_type
 
+    def _filter_attributed_metrics_by_column_names(
+        self,
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        include_column_names: Optional[List[str]],
+        exclude_column_names: Optional[List[str]],
+    ) -> Dict[Domain, Dict[str, ParameterNode]]:
+        def _filter(m: Dict[Domain, Dict[str, ParameterNode]]) -> bool:
+            column_name: str = m.domain_kwargs.column
+            if exclude_column_names and column_name in exclude_column_names:
+                return False
+            if include_column_names and column_name not in include_column_names:
+                return False
+            return True
+
+        domains: Set[Domain] = set(filter(lambda m: _filter(m), attributed_metrics))
+        filtered_attributed_metrics: Dict[Domain, Dict[str, ParameterNode]] = {
+            domain: attributed_metrics[domain] for domain in domains
+        }
+
+        return filtered_attributed_metrics
+
+    def _filter_attributed_metrics_by_metric_name(
+        self,
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        metric_name: str,
+    ) -> Dict[Domain, Dict[str, ParameterNode]]:
+        domain: Set[Domain]
+        filtered_attributed_metrics: Dict[Domain, Dict[str, ParameterNode]] = {}
+        for domain, attributed_metric_values in attributed_metrics.items():
+            if metric_name in attributed_metric_values.keys():
+                filtered_attributed_metrics[domain] = {
+                    metric_name: attributed_metric_values[metric_name]
+                }
+
+        return filtered_attributed_metrics
```

Comment:
We don't always test private methods, but these might be good candidates to run a few cases through to make sure they work both with expected and unexpected input. To check that they work as intended, but also to guard against future changes breaking expected functionality. What do you think?

Reply:
I like it (as long as it isn't a big lift)

Reply:
I will address this in PR: https://github.com/great-expectations/great_expectations/pull/5259